FWCore/Framework/interface/CallbackBase.h

```diff
@@ -81,15 +81,22 @@ namespace edm {
       CallbackBase(CallbackBase&&) = delete;
       CallbackBase& operator=(CallbackBase&&) = delete;
 
+      template <typename ProduceFunctor>  // ProduceFunctor executes TProduceFunc
       WaitingTaskHolder makeProduceTask(oneapi::tbb::task_group* group,
                                         ServiceWeakToken const& serviceToken,
                                         EventSetupRecordImpl const* record,
                                         EventSetupImpl const* eventSetupImpl,
-                                        bool emitPostPrefetchingSignal) {
+                                        bool emitPostPrefetchingSignal,
+                                        ProduceFunctor&& produceFunctor) {
         return WaitingTaskHolder(
             *group,
-            make_waiting_task([this, group, serviceToken, record, eventSetupImpl, emitPostPrefetchingSignal](
-                                  std::exception_ptr const* iException) {
+            make_waiting_task([this,
+                               group,
+                               serviceToken,
+                               record,
+                               eventSetupImpl,
+                               emitPostPrefetchingSignal,
+                               produceFunctor = std::move(produceFunctor)](std::exception_ptr const* iException) {
```

Comment:
This should now be
```suggestion
                               produceFunctor = std::forward(produceFunctor)](std::exception_ptr const* iException) {
```
to allow the perfect forwarding

Reply:
Done. Thanks.